lightning/src/routing/router.rs

```diff
@@ -1113,10 +1113,16 @@ where L::Target: Logger {
 						fees: hop.fees,
 					};
 
-					let reqd_channel_cap = if let Some (val) = final_value_msat.checked_add(match idx {
-						0 => 999,
-						_ => aggregate_next_hops_fee_msat.checked_add(999).unwrap_or(u64::max_value())
-					}) { Some( val / 1000 ) } else { break; }; // converting from msat or breaking if max ~ infinity
+					// We want a value of final_value_msat * ROUTE_CAPACITY_PROVISION_FACTOR but we
+					// need it to increment at each hop by the fee charged at later hops. Further,
+					// we need to ensure we round up when we divide to get satoshis.
+					let channel_cap_msat = final_value_msat
+						.checked_mul(ROUTE_CAPACITY_PROVISION_FACTOR).unwrap_or(u64::max_value())
+						.map(|v| v.checked_add(aggregate_next_hops_fee_msat)).unwrap_or(u64::max_value());
```

Comment:
I think this should be `and_then` otherwise you'll have nested `Option`s.

Reply:
Ah, forgot about that one.